core/lib/Cardano/Address/Script.hs

```diff
@@ -258,77 +321,201 @@ foldScript fn zero = \case
     foldMScripts =
         runIdentity . foldM (\acc -> Identity . foldScript fn acc) zero
 
-
 --
 -- Script validation
 --
 
 -- | Validate a 'Script', semantically
 --
 -- @since 3.0.0
-validateScript :: Script -> Either ErrValidateScript ()
-validateScript = \case
-    RequireSignatureOf (KeyHash bytes) -> do
-        when (BS.length bytes /= credentialHashSize) $ Left WrongKeyHash
+validateScript
+    :: ValidationLevel
+    -> TxValidity
+    -> Script KeyHash
+    -> Either ErrValidateScript ()
+validateScript level interval script = do
+    let validateKeyHash (KeyHash bytes) =
+            (BS.length bytes == credentialHashSize)
+    let allSigs = foldScript (:) [] script
+    unless (L.all validateKeyHash allSigs) $ Left WrongKeyHash
+    unless (requiredValidation interval script)
+        $ Left LedgerIncompatible
+    when (level == RecommendedValidation ) $
+        recommendedValidation script
+
+requiredValidation
+    :: Eq elem
+    => TxValidity
```

Comment:
I am not quite sure about this argument. It is very unlikely that scripts are going to be created with this knowledge in mind. Scripts are typically created well before transactions. So, this is a check we could perform upon submitting a transaction but not before.

Reply:
made it optional. as I think it should stay here, this possibility to check it for client of this library (eg. wallet). But in command line I use Nothing. see HEAD